src/lib/containers/download_list_v2/DownloadListTable.tsx

```diff
@@ -44,9 +52,14 @@ export default function DownloadListTable() {
     fetchNextPage,
     isError,
     error: newError,
-    refetch,
+    refetch
   } = useGetAvailableFilesToDownloadInfinite(sort, filter)
 
+  //SWC-5858: Update the Download List files table when the statistics change
+  useDeepCompareEffect(() => {
+    refetch()
```

Comment:
Instead of recreating the entire component, just monitor the file statistics (deep compare) and refetch the list if anything changes.

Reply:
Probably much more efficient!